k8s/predelete.go

```diff
@@ -0,0 +1,144 @@
+package k8s
+
+import (
+	"context"
+	"tyk/tyk/bootstrap/pkg/constants"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ExecutePreDeleteOperations executes operations needed in pre-delete chart hook one by one.
+func (c *Client) ExecutePreDeleteOperations() error {
+	c.l.Info("Running pre delete hook")
+
+	if err := c.deleteOperatorSecret(); err != nil {
+		return err
+	}
+
+	if err := c.deletePortalSecret(); err != nil {
+		return err
+	}
+
+	if err := c.deleteBootstrappingJobs(); err != nil {
+		return err
+	}
+
+	c.l.Info("Run pre delete hook successfully")
+
+	return nil
+}
+
+// deleteOperatorSecret deletes the Kubernetes secret created specifically for Tyk Operator.
+func (c *Client) deleteOperatorSecret() error {
+	secrets, err := c.clientSet.
+		CoreV1().
+		Secrets(c.appArgs.K8s.ReleaseNamespace).
+		List(context.TODO(), metav1.ListOptions{})
```

Comment:
Can't we directly use Delete function here?

Reply:
implemented as you suggested